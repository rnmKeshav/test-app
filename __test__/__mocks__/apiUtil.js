import apiUtil from "../../src/helpers/apiUtil";

const apiUtilMock = () => {
  //jest.mock("../../src/helpers/apiUtil");
  return {
    // get: {
    //   success: resp => {
    //     apiUtil.get = jest.fn(() => {
    //       return Promise.resolve(resp);
    //     });
    //   },
    //   failure: resp => {
    //     // apiUtil.get = jest.fn(() => {
    //     //   return Promise.reject(resp);
    //     // });
    //     apiUtil.get.mockImplementation(() => {
    //       return Promise.reject(resp);
    //     });
    //   }
    // }

    get: {
      success: resp => {
        apiUtil.get = jest.fn(() => Promise.resolve(resp));
      },

      failure: resp => {
        apiUtil.get = jest.fn(() => Promise.reject(resp));
      }
    }
  };
};

module.exports = { apiUtilMock: apiUtilMock() };
