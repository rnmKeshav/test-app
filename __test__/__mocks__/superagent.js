//import superagent from "superagent";

const mockedSuperagent = jest.mock("superagent", () => {
  return {
    get: jest.fn().mockImplementation(() => {
      return this;
    }),
    query: jest.fn().mockImplementation(() => {
      console.log("query called");
      return this;
    }),
    then: jest.fn().mockImplementation(resp => {
      return this;
    })
  };
});
// jest.mock("superagent");

// const mockedSuperagent = function() {
//   return {
//     get: () => {
//       superagent.get.mockImplementation(function() {
//         console.log("t");
//         return this;
//       });
//     },

//     query: () => {
//       superagent.query.mockImplementation(function() {
//         return this;
//       });
//     },

//     then: resp => {
//       superagent.then.mockImplementation(function() {
//         return resp;
//       });
//     }
//   };
// };

export default mockedSuperagent;
