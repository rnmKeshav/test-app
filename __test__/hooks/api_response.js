const getUserData = () => ({
  login: "rnmKeshav",
  id: 11366455,
  node_id: "MDQ6VXNlcjExMzY2NDU1",
  avatar_url: "https://avatars1.githubusercontent.com/u/11366455?v=4",
  gravatar_id: "",
  url: "https://api.github.com/users/rnmKeshav",
  html_url: "https://github.com/rnmKeshav",
  followers_url: "https://api.github.com/users/rnmKeshav/followers",
  following_url: "https://api.github.com/users/rnmKeshav/following{/other_user}",
  gists_url: "https://api.github.com/users/rnmKeshav/gists{/gist_id}",
  starred_url: "https://api.github.com/users/rnmKeshav/starred{/owner}{/repo}",
  subscriptions_url: "https://api.github.com/users/rnmKeshav/subscriptions",
  organizations_url: "https://api.github.com/users/rnmKeshav/orgs",
  repos_url: "https://api.github.com/users/rnmKeshav/repos",
  events_url: "https://api.github.com/users/rnmKeshav/events{/privacy}",
  received_events_url: "https://api.github.com/users/rnmKeshav/received_events",
  type: "User",
  site_admin: false,
  name: "Keshav kumar",
  company: null,
  blog: "",
  location: null,
  email: null,
  hireable: null,
  bio: null,
  public_repos: 3,
  public_gists: 0,
  followers: 1,
  following: 0,
  created_at: "2015-03-07T16:49:12Z",
  updated_at: "2018-06-20T19:17:08Z"
});

module.exports = {
  getUserData
};
