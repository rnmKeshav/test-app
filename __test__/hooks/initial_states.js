let loggedinUser = {
  isFetching: false,
  isModalOpen: false,
  error: {
    status: false,
    body: {}
  },
  userHandle: "",
  details: {}
};

module.exports = {
  loggedinUser
};
