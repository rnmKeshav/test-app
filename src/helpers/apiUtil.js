import request from "superagent";

/***********
options = {
  endpoint: '/get_users',
  query: {
    a: b
  },
  headers: {
    Content-Type: application/json
  },
  payload: {} // for POST/PATCH/DELETE
}
**********/
const apiRequest = (method, options = {}, host = "https://api.github.com/") => {
  let { endpoint, query = {}, payload = {} } = options;
  let headers = { accept: "application/json" };
  host = host || window.location.origin;
  let url = `${host}${endpoint}`;

  console.log("url", url);
  Object.assign(headers, options.headers);

  let req = request[method.toLowerCase()](url)
    .query(query)
    .set(headers);

  if (method.toLowerCase() !== "get") {
    req.send(payload);
  }

  return new Promise((resolve, reject) => {
    req.end((error, res) => {
      let response;

      if (error) {
        reject(error);
      }

      try {
        if (res && res.text !== "") {
          response = JSON.parse(res.text);
        }
      } catch (e) {
        reject(e);
      }

      resolve(response);
    });
  });
};

const api = {
  get: apiRequest.bind(null, "get"),
  post: apiRequest.bind(null, "post"),
  patch: apiRequest.bind(null, "patch"),
  delete: apiRequest.bind(null, "delete")
};

export default api;
