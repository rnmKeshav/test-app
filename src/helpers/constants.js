const actionTypes = {
  "LOGGEDIN_USER/DATA_FETCH_STARTED": "LOGGEDIN_USER/DATA_FETCH_STARTED",
  "LOGGEDIN_USER/DATA_FETCH_FAILURE": "LOGGEDIN_USER/DATA_FETCH_FAILURE",
  "LOGGEDIN_USER/DATA_FETCH_SUCCESS": "LOGGEDIN_USER/DATA_FETCH_SUCCESS",
  "LOGGEDIN_USER/CHANGE_HANDLE": "LOGGEDIN_USER/CHANGE_HANDLE",
  "LOGGEDIN_USER/UPDATE_TO_LOCALSTORE": "LOGGEDIN_USER/UPDATE_TO_LOCALSTORE"
};

module.exports = {
  actionTypes
};
