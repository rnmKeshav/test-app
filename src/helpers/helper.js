const debounce = (fn, delay) => {
  let timeout;

  return payload => (dispatch, getState) => {
    clearTimeout(timeout);
    timeout = setTimeout(() => {
      dispatch(fn(payload));
    }, delay);
  };
};

module.exports = {
  debounce
};
