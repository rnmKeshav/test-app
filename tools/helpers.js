let parseArr = (arr) => (arg) => 
  arr
    .filter(item => !!item)
    .map(item => item(arg));

module.exports = {
  parseArr
};
