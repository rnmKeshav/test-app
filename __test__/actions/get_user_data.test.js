import getUserData from "../../src/actions/get_user_data";
import { actionTypes } from "../../src/helpers/constants";

import { apiUtilMock } from "../__mocks__/apiUtil";

import { getUserData as userMockData } from "../hooks/api_response";
import storeMocker from "../hooks/store_mocker";

describe("get_user_data action test", () => {
  test("should get user data successfully", () => {
    let store = storeMocker({});
    let userData = userMockData();
    let expectedActions = [
      { type: actionTypes["LOGGEDIN_USER/DATA_FETCH_STARTED"] },
      {
        type: actionTypes["LOGGEDIN_USER/DATA_FETCH_SUCCESS"],
        payload: { data: userData }
      }
    ];

    apiUtilMock.get.success(userData);

    return store.dispatch(getUserData({ handle: "rnmkeshav" })).then(resp => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  test("should handle error", () => {
    let store = storeMocker({});
    apiUtilMock.get.failure({
      message: "something went wrong"
    });

    // For handling error, catching is necessary
    return store.dispatch(getUserData({})).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
