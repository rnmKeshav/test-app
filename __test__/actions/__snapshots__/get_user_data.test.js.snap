// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get_user_data action test should handle error 1`] = `
Array [
  Object {
    "type": "LOGGEDIN_USER/DATA_FETCH_STARTED",
  },
  Object {
    "payload": Object {
      "message": "something went wrong",
    },
    "type": "LOGGEDIN_USER/DATA_FETCH_FAILURE",
  },
]
`;
